write-fonts/src/tables/glyf.rs

```diff
@@ -118,8 +118,14 @@ impl ContourPoint {
         Self::new(point, false)
     }
 
-    fn distance(&self, other: &Self) -> f64 {
-        self.point.distance(other.point)
+    /// Return True if p0, p1 (self) and p2 are collinear and p1 is equidistant from p0 and p2
+    fn is_midpoint_between(&self, p0: &Self, p2: &Self) -> bool {
+        let p1 = self;
+        let p1p0 = p1.point - p0.point;
+        let p2p1 = p2.point - p1.point;
+        // should tolerance be a parameter?
+        let eps = f32::EPSILON as f64;
+        (p1p0.x - p2p1.x).abs() < eps && (p1p0.y - p2p1.y).abs() < eps
```

Comment:
Can this be written more tersely? - the correct mid is (p0+p2) / 2 and we want to know if p1 is within epsilon of it. If we multiply p1 instead of dividing isn't that:

```rust
let delta_mid = p0 + p2 - 2 * p1; // how far from the midpoint we are
mid.x.abs() < f32::EPSILON && mid.y.abs() < f32::EPSILON
```

Reply:
But p0, p1 and P2 are points not vectors, you can't sum them up or multiply by a scalar I think..

Reply:
In looking closer kurbo implements midpoint, subtraction, and multiply by affine so it should have all the parts. Let me try real quick.

Reply:
https://github.com/googlefonts/fontations/pull/474 exhibits; maybe the affine and the tolerance might want to be constants if we keep it. Feel free to merge or ignore.